driver/src/main/java/org/neo4j/driver/internal/net/ChannelFactory.java

```diff
@@ -69,7 +65,7 @@ private static void connect( SocketChannel soChannel, BoltServerAddress address,
         }
         catch ( SocketTimeoutException e )
         {
-            throw new IOException( "Timeout " + timeoutMillis + "ms expired", e );
+            throw new ConnectException( "Timeout " + timeoutMillis + "ms expired" + e.getMessage() );
```

Comment:
Maybe we can keep `IOException`? Javadoc for `ConnectException` says "Typically, the connection was refused remotely (e.g., no process is listening on the remote address/port)" which is a bit different in our case.